### PR TITLE
Enable ctdb test in Aggregates in osd

### PR DESCRIPTION
**Pull request by lilyeyes**
Enable ctdb test in Aggregates in osd

- Related MR for osd: https://gitlab.suse.de/qa-maintenance/qam-openqa-yml/-/merge_requests/784
- Related MR for ow15:  https://gitlab.suse.de/qa-css/openqa_ha_sap/-/merge_requests/771
- Related ticket: https://jira.suse.com/browse/TEAM-9607
- Verification run: 
15sp6: http://openqa.suse.de/tests/15291380#dependencies (all passed)
15sp5: http://openqa.suse.de/tests/15304274#dependencies (all passed) 
15sp4: http://openqa.suse.de/tests/15303718#dependencies (all passed)
15sp3: http://openqa.suse.de/tests/15291705#dependencies (all passed)
15sp2: http://openqa.suse.de/tests/15294390#dependencies (all passed)
12sp5: http://openqa.suse.de/tests/15304261#dependencies (all passed) 
VRs for `VIRTIO_CONSOLE=1`:
15sp5: http://openqa.suse.de/tests/15304760#dependencies (all passed) (VIRTIO_CONSOLE=1)
12sp5: http://openqa.suse.de/tests/15304753#dependencies (all passed) (VIRTIO_CONSOLE=1)

### CODE REVIEW

File: tests/ha/clvmd_lvmlockd.pm [hunk 1]
```diff
@@ -21,9 +21,10 @@ sub run {
     my $lvm_conf = '/etc/lvm/lvm.conf';
     my $lock_mgr = 'clvm';
 
-    # lvmlockd is only available in SLE15+
+    # Only SLE-15+ has support for lvmlockd
+    set_var('USE_LVMLOCKD', 0) if (get_var('USE_LVMLOCKD') and is_sle('<15'));
+
     if (get_var("USE_LVMLOCKD")) {
```
Comment (mpagot) [inline tests/ha/clvmd_lvmlockd.pm:27]: Here the logic was:
"If a user externally (in the JobGroup) try to set `USE_LVMLOCKD` in a test not for 15 --> FAILS at it is not supported"

So configuration was outside the test code, validation only was within the test code

File: schedule/ha/bv/basic_cluster_node.yaml [hunk 2]
```diff
@@ -40,7 +40,7 @@ vars:
   # LVM locking daemon used since SLE 15
   USE_LVMLOCKD: '1'
   USE_SUPPORT_SERVER: '1'
-  VIRTIO_CONSOLE: '0'
+  VIRTIO_CONSOLE: '1'
```
Comment (mpagot) [inline schedule/ha/bv/basic_cluster_node.yaml:43]: Why this change is needed?

File: tests/ha/clvmd_lvmlockd.pm [hunk 3]
```diff
@@ -21,9 +21,10 @@ sub run {
     my $lvm_conf = '/etc/lvm/lvm.conf';
     my $lock_mgr = 'clvm';
 
-    # lvmlockd is only available in SLE15+
+    # Only SLE-15+ has support for lvmlockd
+    set_var('USE_LVMLOCKD', 0) if (get_var('USE_LVMLOCKD') and is_sle('<15'));
```
[the remark this answers is not in the document]
Comment (mpagot) [reply tests/ha/clvmd_lvmlockd.pm:25]: May we survive not having `set_var` ? Two possible approach:
- Do not touch the code here but always set `USE_LVMLOCKD` in all relevant Job group
or alternatively
- get rid of `USE_LVMLOCKD` at all everywhere. Let only the test to decide about it not based on a setting but purely using `is_sles`

File: schedule/ha/bv/basic_cluster_node.yaml (hunk 2 above)
Comment (lilyeyes) [reply schedule/ha/bv/basic_cluster_node.yaml:43]: The `VIRTIO_CONSOLE: '0'` here will overwrite the `VIRTIO_CONSOLE: '1'` in job template yaml file `settings`

File: tests/ha/clvmd_lvmlockd.pm (hunk 3 above)
[the remark this answers is not in the document]
Comment (lilyeyes) [reply tests/ha/clvmd_lvmlockd.pm:25]: This is the easiest way to handle `USE_LVMLOCKD` as is defined in openQA test code `schedule/ha/bv/basic_cluster_node.yaml:  USE_LVMLOCKD: '1'`.
Other wise I have to revise lots of job template files.

[the remark this answers is not in the document]
Comment (lilyeyes) [reply tests/ha/clvmd_lvmlockd.pm:25]: > Now the logic become: "No matter what a user externally (in the JobGroup) try to set `USE_LVMLOCKD` --> disable setting if less than 15"
> 
> So configuration is now both outside the test code and internally, there's no more validation internally as no needed anymore as test is also doing configuration

Yes.

File: tests/ha/clvmd_lvmlockd.pm (hunk 1 above)
Comment (lilyeyes) [reply tests/ha/clvmd_lvmlockd.pm:27]: Yes.

**Comment by alvarocarvajald** [comment]
> VRs for VIRTIO_CONSOLE=1:
15sp5: http://openqa.suse.de/tests/15304760#dependencies (all passed) (VIRTIO_CONSOLE=1)
12sp5: http://openqa.suse.de/tests/15304753#dependencies (all passed) (VIRTIO_CONSOLE=1)

Can you add also verification runs for 15-SP7? Many scenarios there (alpha, beta, delta, gamma and ctdb) also use this schedule, and are running with `VIRTIO_CONSOLE: 0`.

**Comment by lilyeyes** [comment]
> > VRs for VIRTIO_CONSOLE=1:
> > 15sp5: http://openqa.suse.de/tests/15304760#dependencies (all passed) (VIRTIO_CONSOLE=1)
> > 12sp5: http://openqa.suse.de/tests/15304753#dependencies (all passed) (VIRTIO_CONSOLE=1)
> 
> Can you add also verification runs for 15-SP7? Many scenarios there (alpha, beta, delta, gamma and ctdb) also use this schedule, and are running with `VIRTIO_CONSOLE: 0`.

Sure, I forgot sp7. I will do the VRs for sp7 next week.

File: tests/ha/clvmd_lvmlockd.pm (hunk 3 above)
[the remark this answers is not in the document]
Comment (lilyeyes) [reply tests/ha/clvmd_lvmlockd.pm]: I tried the second way (`get rid of USE_LVMLOCKD at all everywhere ...`) : but 
 there are some potential risks such as: https://openqa.suse.de/tests/15318156#step/vg/10 to take care of. The USE_LVMLOCKD is used  in different test modules/schedule-yaml-fles in openQA test code, so the code changes and VRs are big.
Also using `USE_LVMLOCKD`  is better than `is_sle(15+)`.

I will try the first way (set `USE_LVMLOCKD` in all relevant Job groups).

[the remark this answers is not in the document]
Comment (lilyeyes) [reply tests/ha/clvmd_lvmlockd.pm]: Done. Deleted USE_LVMLOCKD from the schedule yaml file and added it to the job groups settings (Please see the related 2 MRs.).

**Comment by lilyeyes** [comment]
Hi @mpagot  & @alvarocarvajald revised PR and MRs according to Michele's comments. Please help to review PR/MRs again, thanks.
More VRs:
- 12sp5: https://openqa.suse.de/tests/15318514#dependencies (passed, USE_LVMLOCKD=0 VIRTIO_CONSOLE=1)
- 15sp7:
  ha_ctdb_node:  http://openqa.suse.de/tests/15318187#dependencies (passed, USE_LVMLOCKD=1 VIRTIO_CONSOLE=1)
  ha_alpha_node: http://openqa.suse.de/tests/15318161#dependencies (failed on bsc#1229901, USE_LVMLOCKD=1 VIRTIO_CONSOLE=1)
  ha_beta_node:  http://openqa.suse.de/tests/15318163#dependencies (failed on bsc#1229901, USE_LVMLOCKD=1 VIRTIO_CONSOLE=1)
  ha_delta_node: http://openqa.suse.de/tests/15318190#dependencies (passed, USE_LVMLOCKD=1 VIRTIO_CONSOLE=1)
  ha_gamma_node: http://openqa.suse.de/tests/15318204#dependencies (passed, USE_LVMLOCKD=1 VIRTIO_CONSOLE=1)

**Comment by alvarocarvajald** [comment]
> Hi @mpagot & @alvarocarvajald revised PR and MRs according to Michele's comments. Please help to review PR/MRs again, thanks. More VRs:
> 
> * 12sp5: https://openqa.suse.de/tests/15318514#dependencies (passed, USE_LVMLOCKD=0 VIRTIO_CONSOLE=1)
> * 15sp7:
>   ha_ctdb_node:  http://openqa.suse.de/tests/15318187#dependencies (passed, USE_LVMLOCKD=1 VIRTIO_CONSOLE=1)
>   ha_alpha_node: http://openqa.suse.de/tests/15318161#dependencies (failed on bsc#1229901, USE_LVMLOCKD=1 VIRTIO_CONSOLE=1)
>   ha_beta_node:  http://openqa.suse.de/tests/15318163#dependencies (failed on bsc#1229901, USE_LVMLOCKD=1 VIRTIO_CONSOLE=1)
>   ha_delta_node: http://openqa.suse.de/tests/15318190#dependencies (passed, USE_LVMLOCKD=1 VIRTIO_CONSOLE=1)
>   ha_gamma_node: http://openqa.suse.de/tests/15318204#dependencies (passed, USE_LVMLOCKD=1 VIRTIO_CONSOLE=1)

Thanks Lili!

**Review by alvarocarvajald — APPROVED** [review]
LGTM

**Review by mpagot — COMMENTED** [review]
LGTM
